### PR TITLE
Add possibility for second level navigation on the Sidebar

### PR DESCRIPTION
**Pull request by afonsodelgado**
### Purpose
This PR proposes second-level navigation on the `Sidebar` component. The main idea is to enable `Sidebar.NavItem` to receive other `Sidebar.NavItem` with a `secondary` style prop as children to have nested navigation for that item.

Sidebar-2nd-Level

### Changes
* Make `Sidebar.NavItem` receive components as children and add a `label` prop to render the item text;
* Add `SubNavList` component as a styled list to hold the second-level navigation components;
* Add an `isArray` type check function to common `util/type-check`;
* Fix tests and some minor styles on `Sidebar.Header`.

### CODE REVIEW

File: src/components/Sidebar/components/SubNavList/utils.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { isArray } from '../../../../util/type-check';
+
+const getSelectedChildIndex = children =>
+  isArray(children) ? children.findIndex(child => child.props.selected) : 0;
```
Comment (connor-baer) [inline src/components/Sidebar/components/SubNavList/utils.js]: `findIndex` is not supported in IE. Luckily, `lodash` provides `findIndex` as well: https://lodash.com/docs/4.17.11#findIndex

Comment (afonsodelgado) [reply src/components/Sidebar/components/SubNavList/utils.js]: Fixed. Thanks a lot!

File: src/components/Sidebar/components/SubNavList/SubNavList.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      background: ${theme.colors.p500};
+      height: ${SUB_NAV_ITEM_HEIGHT}px;
+      border-radius: ${theme.borderRadius.kilo};
+      position: absolute;
+      top: calc(${SUB_NAV_ITEM_HEIGHT}px * ${selectedChildIndex});
+      transition: top ${theme.transitions.default};
+    }
+  `;
+
+const SubNavigationContainer = styled('ul')(
```
Comment (herberthenrique) [inline src/components/Sidebar/components/SubNavList/SubNavList.js]: maybe `styled.ul` ?

File: src/components/Sidebar/components/NavItem/NavItem.spec.js [hunk 3]
```diff
@@ -23,16 +33,15 @@ describe('NavItem', () => {
       expect(actual).toMatchSnapshot();
     });
 
-    it('should render children', () => {
+    it('should render children when selected', () => {
       const wrapper = shallow(
```
Comment (herberthenrique) [inline src/components/Sidebar/components/NavItem/NavItem.spec.js]: We don't use `shallow` anymore, could you change to `mount`?

Comment (herberthenrique) [reply src/components/Sidebar/components/NavItem/NavItem.spec.js]: And this will be applied to all the tests in this file and maybe in all the navigation

**Pull request merged**